aperturedb/ParallelQuery.py

```diff
@@ -0,0 +1,159 @@
+import math
+import time
+from threading import Thread
+
+from aperturedb import ProgressBar
+
+import numpy as np
+
+class ParallelQuery:
+
+    """
+        Parallel and Batch Querier for ApertureDB
+    """
+
+    def __init__(self, db, dry_run=False):
+
+        # db can me None in the case of Downloaders
+        # or other abstractions that want to use the parallel/batching
+        # structures without running queries
+        if db:
+            self.db = db.create_new_connection()
+
+        self.dry_run = dry_run
+
+        self.type = "query"
+
+        # Default Values
+        self.batchsize  = 1
+        self.numthreads = 1
+
+        self.total_queries = 0
+        self.times_arr = []
+        self.queries_time = 0
+        self.error_counter  = 0
+
+        self.loader_filename = "progress.log"
+
+    def  get_times(self):
+
+        return self.times_arr
+
+    def generate_batch(self, data):
+
+        """
+            Here we flatten the individual queries to run them as
```

Comment:
Are there other queries other than the Add stuff in loaders that would be using this ? All the loaders implement their own generate batch. Maybe an example in the docstring would explain the use case clearly.